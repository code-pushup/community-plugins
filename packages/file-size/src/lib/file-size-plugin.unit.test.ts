import { describe, expect, it } from 'vitest';
import { pluginConfigSchema } from '@code-pushup/models';
import fileSizePlugin from './file-size-plugin';

describe('fileSizePlugin', () => {
  it('should create valid file-size plugin configuration', async () => {
    const fileSizePluginConfig = fileSizePlugin({});
    expect(() => pluginConfigSchema.parse(fileSizePluginConfig)).not.toThrow();
    expect(fileSizePluginConfig).toMatchSnapshot();
  });
});
