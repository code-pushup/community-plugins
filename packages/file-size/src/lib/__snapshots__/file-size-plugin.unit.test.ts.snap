// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fileSizePlugin > should create valid file-size plugin configuration 1`] = `
{
  "audits": [
    {
      "description": "file-size audit ",
      "slug": "file-size-audit",
      "title": "file-size Audit",
    },
  ],
  "description": "file-size plugin for Code PushUp",
  "icon": "folder",
  "runner": [Function],
  "slug": "file-size",
  "title": "file-size",
}
`;
